src/InMemoryProvider.ts

```diff
@@ -5,7 +5,7 @@
  *
  * NoSqlProvider provider setup for a non-persisted in-memory database backing provider.
  */
-
+import 'core-js/features/set-immediate';
```

Comment:
I'm curious why you'd prefer to import the polyfill rather than utilize `setTimeout(..., 0)`
From a behavioural standpoint, it should be the same, but ideally we'd remove the existing polyfills for promise finally once it's more broadly adopted.

Reply:
Well nodejs does implement setImmediate and this polyfill takes that into account. Hence I thought it may be better to keep it as set immediate. Additionally the polyfill uses channel messaging to implement set immediate, so it doesn’t get affected by things like timer throttling on browsers